pallets/asset/src/lib.rs

```diff
@@ -2397,7 +2402,10 @@ impl<T: Config> Module<T> {
 
 impl<T: Config> Module<T> {
     fn invalid_granularity(ticker: &Ticker, value: Balance) -> bool {
-        !Self::check_granularity(&ticker, value)
+        if let Some(security_token) = Tokens::get(ticker) {
+            return Self::ensure_token_granular(&security_token, &value).is_err();
+        }
+        true
```

Comment:
```suggestion
        Self::ensure_granular(ticker, value).is_err()
```
Can use the `ensure_granular` method below.